File: app/src/main/java/com/example/androidscript/UITemplate/BlockAdapter.java
package com.example.androidscript.UITemplate;

import androidx.recyclerview.widget.RecyclerView;

import java.util.Vector;

public abstract class BlockAdapter<VH extends RecyclerView.ViewHolder> extends RecyclerView.Adapter<VH> {
    public Vector<Vector<String>> Data;

    public updater onOrderChange;

    public interface updater {
        void swap(int a, int b);
        void delete(int a);
        void insert();
        void self(int index);
    }
}

File: app/src/main/java/com/example/androidscript/uitemplate/BlockAdapter.kt
package com.example.androidscript.uitemplate

import androidx.recyclerview.widget.RecyclerView
import java.util.*

abstract class BlockAdapter<VH : RecyclerView.ViewHolder> : RecyclerView.Adapter<VH>() {
    lateinit var data: Vector<Vector<String>>
    lateinit var onOrderChange: Updater

    interface Updater {
        fun swap(a: Int, b: Int)
        fun delete(a: Int)
        fun insert()
        fun self(index: Int)
    }
}